azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -1375,3 +1408,45 @@ func resourceArmContainerGroupPortsHash(v interface{}) int {
 
 	return hashcode.String(buf.String())
 }
+
+func flattenContainerGroupDnsConfig(input *containerinstance.DNSConfiguration) []interface{} {
+	if input == nil {
+		return nil
+	}
+	outputArr := make([]interface{}, 1)
+	output := make(map[string]interface{})
+
+	if v := input.SearchDomains; v != nil {
+		output["search_domains"] = *v
+	}
+	if v := input.Options; v != nil {
+		output["options"] = *v
+	}
+	if v := input.NameServers; v != nil {
+		output["nameservers"] = *v
+	}
+
+	outputArr = append(outputArr, output)
+	return outputArr
+}
+
+func expandContainerGroupDnsConfig(d *schema.ResourceData) *containerinstance.DNSConfiguration {
+	configRaw := d.Get("dns_config").([]interface{})
+	if len(configRaw) == 0 {
+		return nil
+	}
+	config := configRaw[0].(map[string]interface{})
+
+	ns := []string{}
+	for _, v := range config["nameservers"].([]interface{}) {
+		ns = append(ns, v.(string))
+	}
+
+	dnsConfig := &containerinstance.DNSConfiguration{
+		Options:       utils.String(config["options"].(string)),
+		SearchDomains: utils.String(config["search_domains"].(string)),
+		NameServers:   &ns,
+	}
+
+	return dnsConfig
```

Comment:
Can we simply return the structure literal here:

```suggestion
	return &containerinstance.DNSConfiguration{
		Options:       utils.String(config["options"].(string)),
		SearchDomains: utils.String(config["search_domains"].(string)),
		NameServers:   &ns,
	}
```

Reply:
This seems a bit more than minor, and the example you have provided was a TypeList not a TypeSet. I will attempt to refactor for this, but when I was testing last night it broke...a lot.